packages/core/test/unit/query-generator/attributes-to-sql.test.ts

```diff
@@ -0,0 +1,221 @@
+    expectPerDialect(() => queryGenerator.attributesToSQL({ id: { type: 'INTEGER', unique: true, comment: 'This is my comment' } }), {
+      default: { id: 'INTEGER UNIQUE COMMENT This is my comment' },
+      'mariadb mysql snowflake': { id: `INTEGER UNIQUE COMMENT 'This is my comment'` },
+      mssql: { id: 'INTEGER NULL UNIQUE COMMENT This is my comment' },
+      'sqlite ibmi': { id: 'INTEGER UNIQUE' },
+    });
+  });
+
+  it(`{ id: { type: 'INTEGER', unique: true, comment: 'This is my comment' } }, { context: 'addColumn', key: 'column', table: { schema: 'foo', tableName: 'bar' } }`, () => {
+    expectPerDialect(() => queryGenerator.attributesToSQL({ id: { type: 'INTEGER', unique: true, comment: 'This is my comment' } }, { context: 'addColumn', key: 'column', table: { schema: 'foo', tableName: 'bar' } }), {
+      default: { id: 'INTEGER UNIQUE COMMENT This is my comment' },
+      'mariadb mysql snowflake': { id: `INTEGER UNIQUE COMMENT 'This is my comment'` },
+      postgres: { id: `INTEGER UNIQUE; COMMENT ON COLUMN "foo"."bar"."column" IS 'This is my comment'` },
+      mssql: { id: 'INTEGER NULL UNIQUE COMMENT This is my comment' },
+      'sqlite ibmi': { id: 'INTEGER UNIQUE' },
+    });
+  });
+
+  it(`{ id: { type: 'INTEGER', after: 'Bar' } }`, () => {
+    // @ts-expect-error -- after is not accepted in AttributeOptions so attributesToSQL might need to be typed differently
+    expectPerDialect(() => queryGenerator.attributesToSQL({ id: { type: 'INTEGER', after: 'Bar' } }), {
+      default: { id: 'INTEGER' },
+      'mariadb mysql': { id: 'INTEGER AFTER `Bar`' },
+      mssql: { id: 'INTEGER NULL' },
+      'snowflake ibmi': { id: 'INTEGER AFTER "Bar"' },
+    });
+  });
+
+  // TODO: check what this test is supposed to do and update the possible defaultValue accordingly
+  it(`No empty array as default value for BLOB allowed for some dialects`, () => {
+    expectPerDialect(() => queryGenerator.attributesToSQL({ id: { type: 'BLOB', defaultValue: [] } }), {
+      default: { id: 'BLOB' },
+      'postgres sqlite mssql db2': new Error('Could not guess type of value [] because it is an empty array'),
+    });
+  });
+
+  // TODO: this test is broken for at least DB2. It adds the default value because the data type is a string, which can't have special behaviors.
+  //  change type to an actual Sequelize DataType & re-enable
+  it(`No buffer from empty array as default value for BLOB allowed for some dialects`, () => {
+    expectPerDialect(() => queryGenerator.attributesToSQL({ id: { type: 'BLOB', defaultValue: Buffer.from([]) } }), {
+      default: { id: 'BLOB' },
+      postgres: { id: 'BLOB DEFAULT \'\\x\'' },
+      mssql: { id: 'BLOB DEFAULT 0x' },
+      sqlite: { id: 'BLOB DEFAULT X\'\'' },
+      db2: { id: 'BLOB DEFAULT BLOB(\'\')' },
+    });
+  });
+
+  // TODO: check what this test is supposed to do and update the possible defaultValue accordingly
```

Comment:
Same remarks as the above three review notes:

Because the type is a string instead of an instanceof DataType, no special behavior can occur. We use string data types as-is, it's the equivalent of using a literal()
This test should use `DataTypes.TEXT` instead

An array is also not a good type to use to test that the default value is not added: A string would be a better value